src/libmongoc/doc/installing.rst

```diff
@@ -141,162 +186,133 @@ Additionally, it is possible to build only libbson by setting the ``-DENABLE_MON
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-  $ xcode-select --install
+.. parsed-literal::
 
-The ``cmake`` utility is also required. First `install Homebrew according to its instructions <https://brew.sh/>`_, then::
+  $ cmake --build .
+  $ sudo cmake --build . --target install
 
-  $ brew install cmake
+(Note that the ``sudo`` command may not be applicable or available depending on the configuration of your system.)
 
-Download the latest release tarball:
+In the above commands, the first relies on the default target which builds all configured components.  For fine grained control over what gets built, the following command can be used (for Ninja and Makefile-based build systems) to list all available targets:
 
 .. parsed-literal::
 
-  $ curl -LO https://github.com/mongodb/mongo-c-driver/releases/download/x.y.z/mongo-c-driver-x.y.z.tar.gz
-  $ tar xzf mongo-c-driver-x.y.z.tar.gz
-  $ cd mongo-c-driver-x.y.z
+  $ cmake --build . help
+
+Building on Windows with Visual Studio
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once the project files are generated, the project can be opened directly in Visual Studio or compiled from the command line.
 
-Build and install the driver:
+Let's build using the CMake build tool mode:
 
 .. code-block:: none
 
-  $ mkdir cmake-build
-  $ cd cmake-build
-  $ cmake -DENABLE_AUTOMATIC_INIT_AND_CLEANUP=OFF ..
-  $ make install
+  $ cmake --build . --config RelWithDebInfo
 
-All of the same variations described above (e.g., building only libbson, building documentation, etc.) are available when building on macOS.
+Visual Studio's default build type is ``Debug``, but we recommend a release build with debug info for production use. Now that libmongoc and libbson are compiled, let's install them. Components will be installed to the path specified by ``CMAKE_INSTALL_PREFIX``.
 
-.. _build-on-windows:
+.. code-block:: none
 
-Building on Windows with Visual Studio
---------------------------------------
+  $ cmake --build . --config RelWithDebInfo --target install
 
-Building on Windows requires Windows Vista or newer and Visual Studio 2010 or newer. Additionally, ``cmake`` is required to generate Visual Studio project files.
+You should now see libmongoc and libbson installed in ``C:\mongo-c-driver``
 
-Let's start by generating Visual Studio project files. The following assumes we are compiling for 64-bit Windows using Visual Studio 2015 Express, which can be freely downloaded from Microsoft. We will be utilizing ``cmake``'s out-of-source build feature to keep build artifacts separate from source files.
+For Visual Studio 2019 (16.4 and newer), this command can be used to list all available targets:
 
 .. parsed-literal::
 
-  $ cd mongo-c-driver-x.y.z
-  $ mkdir cmake-build
-  $ cd cmake-build
-  $ cmake -G "Visual Studio 14 2015 Win64" \\
-      "-DCMAKE_INSTALL_PREFIX=C:\\mongo-c-driver" \\
-      "-DCMAKE_PREFIX_PATH=C:\\mongo-c-driver" \\
-      ..
+  $ cmake --build . /targets
```

Comment:
I tested this locally. It requires an extra `--`, but I don't think this lists cmake targets.

```
cmake --build . -- /targets
```

Here is a subset of the output
```
CppClean
BeforeCppClean
AfterCppClean
CoreCppClean
PGInstrumentedClean
InitGetClCommandLines
FixupReferences
GetClCommandLines
GetProjectDirectories
GetGeneratedFiles
GetProjectReferencesInfo
GetAssemblyReferences
GetWinMDReferences
GetComReferences
GetSDKReferences
GetProjectReferences
GetForeignReferences
GetResolvedReferences
_PrintPropertyPages
_PrintIntellisenseInfo
_ValidatePathInfo
_PerfIntellisenseInfo
_PrintResolvedProjectReferences
SetTelemetryEnvironmentVariables
```

I believe those are msbuild targets based off of https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-build-specific-targets-in-solutions-by-using-msbuild-exe?view=vs-2019

Perhaps we should just leave the instruction to look at  `*.vcxproj` files as the list of build targets.

Reply:
While CMake has a notion of targets, it creates any specified targets in the chosen build system.  Most build systems have default or built in targets and that seems to be the case here.  In any event, I've left it for now (though I fixed the command to add in the `--` that I missed).  If you still feel strongly like this won't be too helpful, then I'll remove it.

Reply:
True, users could still find this helpful then. This is fine with me.